src/Solvers/fft_based_poisson_solver.jl

```diff
@@ -32,38 +55,47 @@ function FFTBasedPoissonSolver(arch, grid, planner_flag=FFTW.PATIENT)
 end
 
 """
-    solve_poisson_equation!(solver)
+    solve!(x, poisson_solver, b, r=0)
 
-Solves Poisson's equation ∇²ϕ = RHS where `RHS = solver.storage` using periodic or staggered
-Neumann boundary conditions as determined by the `solver.grid`. `solver.storage` will be mutated
-to contain the solution.
+Solves the "screened" poisson equation
+
+```math
+(∇² + m) ϕ = b,
+```
+
+using periodic or Neumann boundary conditions, where ``m`` is a number.
```

Comment:
a real number, if that's true

Reply:
Hmm now I am actually not 100% sure about this. We actually use the complex fft for this case and manually zero out / neglect the complex component for the pressure solve. This may actually work for complex `m`? Unless I'm missing something.

Reply:
Ah, well in that case keep it as is.  Thanks for the info.

Reply:
Are `φ`, `b` real?

Reply:
No.